### DISCUSSION TITLE
module 'BLE' not found & module 'MI32' not found

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#21340] [Q&A] [answered]
Hi, I found [this](https://tasmota.github.io/docs/Bluetooth_MI32/#berry-support)

but if i triy to import this modules i got the error:

![grafik](https://github.com/arendst/Tasmota/assets/16272493/6f006b02-60ac-442e-a6c5-10df1782541b)

![grafik](https://github.com/arendst/Tasmota/assets/16272493/8dd8f453-29d6-4a3d-9737-9f3d292cfb07)

**Comment by Staars** [comment, marked as answer]
If you want to try out the MI32 legacy driver, set the build environment like that:
```
build_flags                 = ${env:tasmota32_base.build_flags}
                              -DFIRMWARE_BLUETOOTH
                              -DUSE_MI_EXT_GUI
                              -DOTA_URL='""'
```

**Comment by ChrisVonFrosta** [comment]
Thank you alot! is there any other option to parse several of the same sensors at the same time?
actually, i read one sensor by adding a rule with `tasmota.add_rule("ATC99dd7f", pars_humidity)`.
It is fine for one sensor but i would like to read some more of this sensors at the same time.

**Reply by sfromis** [reply]
That way is in fact not allowing you to *read* the sensors, but instead will be invoked *when* the sensor decided to send values. Depending on your exact JSON payloads, you *can* set up a `tasmota.add_rule` to react to other sensors, but still only at the time of the sensor volunteering data.

**Reply by ChrisVonFrosta** [reply]
the sensors send its values every 10 second.
looks like this:
```
MQT: tele/tasmota_D503A8/SENSOR = {"Time":"2024-05-07T17:18:34","ATC99dd7f":{"alias":"Glas1","mac":"a4c13899dd7f","Temperature":21.9,"Humidity":67.8,"DewPoint":15.7,"Btn":1,"Battery":4,"RSSI":-101},"TempUnit":"C"}
```

i hoped i could trigger my parsing by "/SENSOR" like:
`tasmota.add_rule("SENSOR", pars_humidity)`

if not i will try the build flags

**Reply by sfromis** [reply]
If you have that specific payload, and your `pars_humidity` is being called due to having `tasmota.add_rule("ATC99dd7f", pars_humidity)`, it looks like the name is misleading.

What I mean is that such a Berry rule should receive the whole object as one parameter, and then you can access each value with code like `sensors['Humidity']`, `sensors['Temperature']` etc, assuming that you have `pars_humidity(sensors)` to get the argument.

And if you have a third argument (the second argument is the trigger), that is a map of the whole sensors payload, including all objects.
